### DISCUSSION TITLE
Drag'n Drop TableRow

### DISCUSSION THREAD

**Discussion by NeoCoderMatrix86** [Megabit/Blazorise#1795] [Q&A]
Hello,
I'm using Blazorise inside a little project (https://neocodermatrix86.github.io/AudioCuesheetEditor/).
Is it possible to have TableRows "draggable" and have a drag and dropped event? It would be used for reordering the table rows by the user.
Thanks for your help :)!

**Comment by stsrki** [comment]
Wow, a very nice app you got there!

Drag is currently not possible out of the box. You can try just adding events directly to the component. Blazorise support attribute splitting so they should be picked up.

Example

```cshtml
<TableRow @ondragstart="@OnDragStartHandler">
```

Btw, once your app is ready, can I announce it to the public? Just say when ;)

**Reply by NeoCoderMatrix86** [reply]
Ok, thanks, I'll try the native ondragstart. Thanks for your question, I would be glad to be mentioned by you :). It's prodution ready and will be developed further :).

**Reply by stsrki** [reply]
> Thanks for your question, I would be glad to be mentioned by you :). It's prodution ready and will be developed further :).

So I can already mention it?

**Reply by NeoCoderMatrix86** [reply]
Yes, you can :). Thanks again and for your help also. Great framework 👍 :)

**Reply by NeoCoderMatrix86** [reply]
So I tried a bit around and currently I'm stuck. I need to have a @ondragstart which prevents default. But when I enter "@ondragstart="() => TrackTableRowOnDragStart(track) @ondragstart:preventDefault" I can not compile. With default blazor this is possible. Do you have an Idea, why this fails?

**Reply by NeoCoderMatrix86** [reply]
Basically I'm trying to implement something like this: https://github.com/phuoc-ng/html-dom/blob/master/demo/drag-and-drop-table-row/index.html
Moving a Row and afterwards recalculating stuff in an "HandleDrop" method.

**Reply by stsrki** [reply]
You're right, it's not possible to define `@ondragstart:preventDefault`. Can you open a new ticket and I can try adding new events to table component(s).

**Reply by NeoCoderMatrix86** [reply]
I opened Ticket #1801. I will work on this the next days, didn't quite get it till now, since I'm stuck a bit in CSS and javascript.

**Comment by stsrki** [comment]
Some suggestions for your app. You can use the new Blazorise localization. It will work better than native Blazor and will not require a page refresh.

**Reply by NeoCoderMatrix86** [reply]
Ok, thanks for your reply, I'll have a look into the Blazorise localization :).

**Reply by NeoCoderMatrix86** [reply]
Do you have a simple example on how to set up Blazorise localization?

**Reply by stsrki** [reply]
You can see an example in the demo https://github.com/stsrki/Blazorise/blob/master/Demos/Blazorise.Demo/Layouts/MainLayout.razor#L257

And in the documentation https://blazorise.com/docs/helpers/localization/

**Reply by NeoCoderMatrix86** [reply]
Good starting point. But it is not mentioned how to set it up in Program.cs and how I get it to use the Resource files?! Also the "ITextLocalizerService" is not known in my project, so do I need to add another package?

**Reply by stsrki** [reply]
You don't need to set up anything in Program.cs or Startup.cs. Everything is already done by Blazorise. You just need to inject to use it `[Inject] protected ITextLocalizerService LocalizationService { get; set; }`. And change the culture by `LocalizationService.ChangeLanguage( "es" );`.

ITextLocalizerService is under the `Blazorise.Localization` namespace so just add the `@using Blazorise.Localization`.

**Reply by NeoCoderMatrix86** [reply]
Ok, I found the problem: I use Version 0.9.2.5, the localization is only available in Version 0.9.3. Thanks for your support, I will change it the next days :).
